spec/registry.xml

```diff
@@ -135,7 +135,79 @@
         </description>
       </option>
     </formatSignature>
+  </function>
+
+  <function name="date">
+    <description>
+      Date formatting
+    </description>
+
+    <formatSignature>
+      <input validationRule="iso8601"/>
+      <option name="style" values="full long medium short">
+        <description>
+          The predefined date formatting style to use.
+        </description>
+      </option>
+      <option name="calendar" values="buddhist chinese coptic dangi ethioaa ethiopic gregory hebrew indian islamic islamic-umalqura islamic-tbla islamic-civil islamic-rgsa iso8601 japanese persian roc">
+        <description>
+          Calendar to use.
+        </description>
+      </option>
+      <option name="numberingSystem" values="arab arabext bali beng deva fullwide gujr guru hanidec khmr knda laoo latn limb mlym mong mymr orya tamldec telu thai tibt">
+        <description>
+          Numbering system to use.
+        </description>
+      </option>
+      <option name="timeZone" validationRule="timeZoneId">
+        <description>
+          The time zone to use.
+          The only value implementations must recognize is "UTC";
+          the default is the runtime's default time zone.
+          Implementations may also recognize the time zone names of the IANA time zone database,
+          such as "Asia/Shanghai", "Asia/Kolkata", "America/New_York".
+        </description>
+      </option>
+    </formatSignature>
+  </function>
+
+  <function name="time">
+    <description>
+      Time formatting
+    </description>
 
+    <formatSignature>
+      <input validationRule="iso8601"/>
+      <option name="style" values="full long medium short">
+        <description>
+          The predefined time formatting style to use.
+        </description>
+      </option>
+      <option name="calendar" values="buddhist chinese coptic dangi ethioaa ethiopic gregory hebrew indian islamic islamic-umalqura islamic-tbla islamic-civil islamic-rgsa iso8601 japanese persian roc">
+        <description>
+          Calendar to use.
+        </description>
+      </option>
+      <option name="numberingSystem" values="arab arabext bali beng deva fullwide gujr guru hanidec khmr knda laoo latn limb mlym mong mymr orya tamldec telu thai tibt">
+        <description>
+          Numbering system to use.
+        </description>
+      </option>
+      <option name="timeZone" validationRule="timeZoneId">
+        <description>
+          The time zone to use.
+          The only value implementations must recognize is "UTC";
+          the default is the runtime's default time zone.
+          Implementations may also recognize the time zone names of the IANA time zone database,
+          such as "Asia/Shanghai", "Asia/Kolkata", "America/New_York".
+        </description>
+      </option>
+      <option name="hourCycle" values="h11 h12 h23 h24">
+        <description>
+          The hour cycle to use.
+        </description>
+      </option>
```

Comment:
This should probably be `hour12` not `hourCycle`

Reply:
Why?

FWIW, I've always found 402's option name `hour12` to be weird. It only describes one or maybe two of the options (and the more common override direction is from 12 to 24 hour)

Reply:
`hour12` sounds like a boolean.

CLDR and the `-u-` extension have 4 possible values: `h11`, `h12`, `h23`, `h24`.
So `hour12` as the name of the option is problematic (can I do hour12=h11 ?)

Also `hourCycle` can be "default" or "auto" (meaning the default for the given locale).
Which again don't work well with a boolean.

Reply:
`hour12` means you want AM/PM or not, but it doesn't express a preference between h11/h12 or between h23/h24, instead leaving that decision up to the locale. It could be the case that a locale prefers h11 for its AM/PM format, but it defaults to h23, for example.

Reply:
`hourCycle` is an _override_ option. The default is `auto` but there is no need to specify `auto`. You only need the cycle to change it.

I could see adding `hour12` or `hour24` as a shorthand to coerce locales from `11/`12` to use `23`/`24`. This has a nice "skeleton-y" quality about it, since it doesn't require users to pass a _specific_ `hc` when it should partly depend on the locale when coercing the other behavior.